File: C#/Nevera.cs
using System;

namespace poocorredor
{
    //Aca vamos a ver un ejemplo de clase y uno de Herencia
    public class Nevera : Electrodomestico
    {
        private double peso;
        private double temperatura;
        private bool sirve;

        //Metodo constructor
        public Nevera(double peso, double temperatura, bool sirve)
        {
            this.peso = peso;
            this.temperatura = temperatura;
            this.sirve = sirve;
        }
        public void ahorrarEnergia(double temperatura)
        {
            if (sirve == false)
            {
                Console.WriteLine("No sirve mijo");
            }
            else if (temperatura < 0)
            {
                encender(false);
            }
            else
            {
                encender(true);
            }
        }

        public override void encender(bool encendido)
        {
            if (encendido == true)
            {
                Console.WriteLine("Se enciende");
            }
            else
            {
                Console.WriteLine("se apaga");
            }
        }

        public override void funcionar()
        {
            Console.WriteLine("Funciono");
        }
    }
}

File: JS/Nevera.js
const Electrodomestico = require("./Electrodomestico");

//Aca vamos a ver un ejemplo de clase y uno de Herencia
class Nevera extends Electrodomestico {
    constructor(peso, temperatura, sirve) {
        super();
        this.peso = peso;
        this.temperatura = temperatura;
        this.sirve = sirve;
    }

    //Metodo para ahorrar energia :D
    ahorrarEnergia(temperatura) {
        if (this.sirve === false) {
            console.log("No sirve");
        } else if (temperatura < 0) {
            this.encender(false);
        } else {
            this.encender(true);
        }
    }
    //Metodos "abstractos"
    encender(encendido) {
        if (encendido === true) {
            console.log("Se enciende");
        } else {
            console.log("se apaga");
        }
    }
    funcionar() {
        console.log("Funciono");
    }
}

module.exports = Nevera;
